### PR TITLE
feat: connect journey tiered data

### PR DESCRIPTION
**Pull request by dyersituations**
https://kiva.atlassian.net/browse/MP-966

- The badge section now shows the correct badge icon: in-progress tier or next tier if no progress
- Badge journey now shows the correct badge icons per tier
- Badge journey now shows the level names from Contentful
- Updated stories with correct data format
- Fixed issue with badge modal opening to incorrect state
- Fixed size of locked badge icon
- Resolved Vue warning related to badge title prop being `null`
- Removed `isBadgeImageLoaded` since images are now the same square size and smaller + faster loading
- Generally cleaned up how badge data is combined and passed from the MyKiva page
- Fixed issues with the small progress circle in the badge journey
- Entire badge is now clickable in MyKiva badge section

image

image

### CODE REVIEW

File: src/components/MyKiva/BadgeInProgress.vue [hunk 1]
```diff
@@ -200,11 +176,11 @@ const onInteractCarousel = interaction => {
 	$kvTrackEvent('carousel', 'click-carousel-horizontal-scroll', interaction);
 };
 
-const badgeName = computed(() => props.badge?.fields?.challengeName ?? '');
+const badgeName = computed(() => props.badge?.challengeName ?? '');
 // TODO: Replace once the work of using the icon of the correct in-progress badge level is complete
 const badgeImage = computed(() => props.badge?.fields?.badgeImage?.fields?.file?.url ?? '');
 const badgeLevel = computed(() => props.badge?.fields?.level ?? '');
-const totalProgress = computed(() => props.badge?.totalProgressToAchievement ?? 0);
+const totalProgress = computed(() => props.badge?.achievementData?.totalProgressToAchievement ?? 0);
```
Comment (roger-in-kiva) [inline src/components/MyKiva/BadgeInProgress.vue:183]: Do we need reactive props here?

Comment (dyersituations) [reply src/components/MyKiva/BadgeInProgress.vue:183]: Probably, since the modal can be re-opened with another badge.

Comment (roger-in-kiva) [reply src/components/MyKiva/BadgeInProgress.vue:183]: Same for other cases...

Comment (dyersituations) [reply src/components/MyKiva/BadgeInProgress.vue:183]: Right, I'm assuming we generally want reactivity here for when the badge modal is re-opened with another badge.

Comment (roger-in-kiva) [reply src/components/MyKiva/BadgeInProgress.vue:183]: It's some upcoming work for that?

Comment (dyersituations) [reply src/components/MyKiva/BadgeInProgress.vue:183]: What do you mean, these fields are already reactive, `computed` based on `props`?

Comment (roger-in-kiva) [reply src/components/MyKiva/BadgeInProgress.vue:183]: True! I mean if passing toRefs is needed...

Comment (dyersituations) [reply src/components/MyKiva/BadgeInProgress.vue:183]: It shouldn't be needed, since we're not deconstructing the props.

**Pull request merged**